src/bin/clapper-app/clapper-app-uri-dialog.c

```diff
@@ -25,30 +25,27 @@
 #include "clapper-app-utils.h"
 
 static void
-_entry_text_changed_cb (GtkEntry *entry,
-    GParamSpec *pspec G_GNUC_UNUSED, AdwMessageDialog *dialog)
+_entry_text_changed_cb (AdwEntryRow *entry,
+    GParamSpec *pspec G_GNUC_UNUSED, AdwAlertDialog *dialog)
 {
-  GtkEntryBuffer *buffer = gtk_entry_get_buffer (entry);
-  guint text_length = gtk_entry_buffer_get_length (buffer);
+  const gchar *text = gtk_editable_get_text (GTK_EDITABLE (entry));
   gboolean enabled = FALSE;
 
-  if (text_length > 0) {
-    const gchar *text = gtk_entry_buffer_get_text (buffer);
+  if (strlen (text) > 0) {
     enabled = (text && gst_uri_is_valid (text));
   }
 
-  adw_message_dialog_set_response_enabled (dialog, "add", enabled);
+  adw_alert_dialog_set_response_enabled (dialog, "add", enabled);
 }
 
 static void
-_open_uri_cb (AdwMessageDialog *dialog, GAsyncResult *result, GtkApplication *gtk_app)
+_open_uri_cb (AdwAlertDialog *dialog, GAsyncResult *result, GtkApplication *gtk_app)
 {
-  const gchar *response = adw_message_dialog_choose_finish (dialog, result);
+  const gchar *response = adw_alert_dialog_choose_finish (dialog, result);
 
   if (strcmp (response, "add") == 0) {
-    GtkWidget *extra_child = adw_message_dialog_get_extra_child (dialog);
-    GtkEntryBuffer *buffer = gtk_entry_get_buffer (GTK_ENTRY (extra_child));
-    const gchar *text = gtk_entry_buffer_get_text (buffer);
+    GtkWidget *extra_child = gtk_list_box_row_get_child (gtk_list_box_get_row_at_index (GTK_LIST_BOX (adw_alert_dialog_get_extra_child (dialog)), 0));
```

Comment:
```suggestion
    GtkWidget *extra_child = adw_alert_dialog_get_extra_child (dialog);
    GtkListBoxRow *list_row = gtk_list_box_get_row_at_index (GTK_LIST_BOX (extra_child), 0);
    GtkWidget *entry_row = gtk_list_box_row_get_child (list_row);
```

Reply:
While this creates more lines, its also clearer to understand how widget hierarchy is placed here, thus less prone to mistakes in future.

Reply:
(this was also wrong :sweat: _get_row_at_index returned the EntryRow so we didn't need the child)